tensorflow/compiler/mlir/xla/transforms/materialize_broadcasts.cc

```diff
@@ -296,8 +296,18 @@ struct CompareWithBroadcastConvert : public OpRewritePattern<CompareOp> {
                                 PatternRewriter &rewriter) const override {
     Value new_lhs;
     Value new_rhs;
-    if (!CreateBroadcastsForBinaryOp(op, &rewriter, &new_lhs, &new_rhs)) {
-      return failure();
+
+    auto op_ranked_type = op.getType().template dyn_cast<RankedTensorType>();
```

Comment:
Could you factor this out into a shared helper instead of replicating the code here?